examples/step-67b/step-67b.cc

```diff
@@ -0,0 +1,1576 @@
+                      }
+                  }
+
+                internal::FEFaceNormalEvaluationImpl<dim,
+                                                     n_points_1d - 1,
+                                                     VectorizedArray<Number>>::
+                  template interpolate_quadrature<false, true>(
+                    dim + 2,
+                    data.get_shape_info(),
+                    phi_m.begin_values(),
+                    phi.begin_values(),
+                    false,
+                    face);
+              }
+
+            for (unsigned int q = 0; q < phi.static_n_q_points; ++q)
+              {
+                const auto factor = VectorizedArray<Number>(1.0) / phi.JxW(q);
+                for (unsigned int c = 0; c < dim + 2; ++c)
+                  phi.begin_values()[c * phi.static_n_q_points + q] =
+                    phi.begin_values()[c * phi.static_n_q_points + q] * factor;
+              }
+
+            internal::FEEvaluationImplBasisChange<
+              dealii::internal::EvaluatorVariant::evaluate_evenodd,
+              internal::EvaluatorQuantity::hessian,
+              dim,
+              degree + 1,
+              n_points_1d,
+              VectorizedArray<Number>,
+              VectorizedArray<Number>>::do_backward(dim + 2,
+                                                    data.get_shape_info()
+                                                      .data[0]
+                                                      .inverse_shape_values_eo,
+                                                    false,
+                                                    phi.begin_values(),
+                                                    phi.begin_dof_values());
+
+            // RK Stage
+            {
+              const Number ai = factor_ai;
+              const Number bi = factor_solution;
+
+              if (ai == Number())
+                {
+                  for (unsigned int q = 0; q < phi.static_dofs_per_cell; ++q)
+                    phi.begin_dof_values()[q] = bi * phi.begin_dof_values()[q];
+                  phi.distribute_local_to_global(solution);
+                }
+              else
+                {
+                  if (stage != 0)
+                    phi_temp.read_dof_values(solution);
+
+                  for (unsigned int q = 0; q < phi.static_dofs_per_cell; ++q)
+                    {
+                      const auto K_i = phi.begin_dof_values()[q];
+
+                      phi.begin_dof_values()[q] =
+                        phi_temp.begin_dof_values()[q] + (ai * K_i);
+
+                      phi_temp.begin_dof_values()[q] += bi * K_i;
+                    }
+                  phi.set_dof_values(dst);
+                  phi_temp.set_dof_values(solution);
+                }
+            }
+          }
+      },
+      vec_ki,
+      current_ri,
+      true,
+      MatrixFree<dim, Number>::DataAccessOnFaces::values);
+  }
```

Comment:
This function is new. It merges `local_apply_inverse_mass_matrix()`, `local_apply_cell()`, `local_apply_face()`, and `local_apply_boundary_face()`.